include/TGUI/Widget.hpp

```diff
@@ -505,6 +505,24 @@ namespace tgui
         Widget::Ptr getToolTip() const;
 
 
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        /// @brief Changes the name of a widget
+        ///
+        /// @param name    New name for the widget
+        ///
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        void setWidgetName(const std::string& name);
+
+
+        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        /// @brief Returns the name of a widget
+        ///
+        /// @return Name of the widget or an empty string when the widget didn't exist or wasn't given a name
```

Comment:
Copy-paste issue: the widget always exists if this function can be called. So it's only empty if it wasn't given a name